sdk/core/azure-core/inc/azure/core/url.hpp

```diff
@@ -239,5 +239,55 @@ namespace Azure { namespace Core {
      * @return Absolute URL with URL-encoded query parameters.
      */
     std::string GetAbsoluteUrl() const;
+
+    /**
+     * @brief Compares with another instance of Url for equality.
+     * @param other another instance of Url.
+     * @return `true` if values of two Url instances are equal, `false` otherwise.
+     *
+     */
+    bool operator==(Url const& other) const
+    {
+      if (this == &other)
+      {
+        return true;
+      }
+
+      if (m_port != other.m_port
+          || m_encodedQueryParameters.size() != other.m_encodedQueryParameters.size()
+          || m_scheme != other.m_scheme || m_host != other.m_host)
+      {
+        return false;
+      }
+
+      size_t const lhsPathLength = m_encodedPath.length();
+      size_t const rhsPathLength = other.m_encodedPath.length();
+
+      size_t const lhsPathStart = (lhsPathLength > 0 && m_encodedPath[0] == '/') ? 1 : 0;
+      size_t const rhsPathStart = (rhsPathLength > 0 && other.m_encodedPath[0] == '/') ? 1 : 0;
```

Comment:
Nit: Add a comment to clarify why its safe to trim the '/' in the comparison.

Reply:
Thanks! I added the comment, and I also added a comment about equality meaning string equality.